pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -154,6 +154,12 @@ func (c *ClusterController) configureExternalCephCluster(cluster *cluster) error
 		if err != nil {
 			return errors.Wrap(err, "failed to get external ceph mon version")
 		}
+		// Create exporter Secret
+		err = nodedaemon.CreateExporterSecret(c.context, cluster.ClusterInfo)
```

Comment:
The challenge is that this will only succeed if the external cluster is using a keyring with sufficient privileges. In most scenarios this would fail since the keyring is just minimal and not an admin key. IIRC the external cluster tests in the CI are just using the admin key, so the CI must not have hit this issue. We could add a check here that only creates the exporter service if the `client.admin` key is set. @yaguangtang Are you using the admin keyring in your external cluster?
```suggestion
		if cluster.ClusterInfo.CephCred.Username == client.AdminUsername {
			err = nodedaemon.CreateExporterSecret(c.context, cluster.ClusterInfo)
```

Reply:
Alternatively, we may need to disable the exporter for external clusters.

Reply:
There is an previous fix  to use non admin keyring for exporters https://github.com/rook/rook/pull/13239 , from my actual environment, in an external cluster, only rgw is in k8s , so only rgw related metrics are collected by this exporter.

Reply:
thanks for explaination, PR updated.